os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -0,0 +1,159 @@
+extern uint32_t __ksram_segment_start__[];
+extern uint32_t __ksram_segment_size__[];
+
+/****************************************************************************
+ * Private variables
+ ****************************************************************************/
+static struct mm_partition_s g_mempartitions[CONFIG_NUM_APPS];
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: mm_initialize_partitions
+ *
+ * Description:
+ * This function is called during os boot to initialize the memory partitions.
+ *
+ * Parameters:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+void mm_initialize_partitions(void)
+{
+	int i = 0;
+	uint32_t start;
+	uint32_t size;
+
+	start = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	size = (REGION_SIZE - start) / CONFIG_NUM_APPS;
+	for (i = 0; i < CONFIG_NUM_APPS; i++)
+	{
+		g_mempartitions[i].start = start;
+		g_mempartitions[i].size = size;
+		g_mempartitions[i].status = MM_PART_FREE;
+		start += size;
+	}
+}
+
+/****************************************************************************
+ * Name: mm_allocate_partition
+ *
+ * Description:
+ * This function is called to allocate a user app partition in which the app
+ * will be loaded by the loader.
+ *
+ * Parameters:
+ *   start_addr : OUT param to hold start address of allocated partition
+ *   size	: OUT param to hold size of allocated partition
+ *
+ * Return Value:
+ *   success: Zero value if partition was successfuly allocated.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int8_t mm_allocate_partition(uint32_t **start_addr, uint32_t *size)
+{
+	int i;
+
+	for (i = 0; i < CONFIG_NUM_APPS; i++)
+	{
+		if (g_mempartitions[i].status == MM_PART_FREE)
+		{
+			g_mempartitions[i].status = MM_PART_USED;
+			*start_addr = (uint32_t *)g_mempartitions[i].start;
+			*size = g_mempartitions[i].size;
+			return OK;
+		}
+	}
+
+	return ERROR;
+}
+
+/****************************************************************************
+ * Name: mm_free_partition
+ *
+ * Description:
+ * This function is called to free a partition at the time of exiting or
+ * killing an application.
+ *
+ * Parameters:
+ *   address: The start address of the partition to free
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+void mm_free_partition(uint32_t address)
+{
+	int i;
```

Comment:
same, Please add valid address check

Reply:
The address value is only used to compare against the table of partitions. There is no valid address range to check for.

Reply:
not even NULL check ?

Reply:
No. NULL check is required only to avoid indirection of a NULL pointer.  We are not doing that here.